### PR TITLE
cmake: Windows fixes

### PR DESCRIPTION
**Pull request by Orphis**

### CODE REVIEW

**Comment by Orphis** [comment]
@ligfx

File: CMake/CCache.cmake [hunk 1]
```diff
@@ -1,7 +1,6 @@
-find_program(CCACHE_BIN ccache)
+find_program(CCACHE_BIN NAMES ccache sccache)
 if(CCACHE_BIN)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_BIN})
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ${CCACHE_BIN})
```
Comment (iwubcode) [inline CMake/CCache.cmake]: Just curious, why isn't this needed?

File: CMake/FlagsOverride.cmake [hunk 2]
```diff
@@ -0,0 +1,7 @@
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  foreach(f CMAKE_C_FLAGS_DEBUG CMAKE_CXX_FLAGS_DEBUG CMAKE_C_FLAGS_RELWITHDEBINFO CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+	  if("${${f}}" MATCHES "/Zi")
+		  string(REGEX REPLACE "/Zi" "/Z7" "${f}" "${${f}}")
```
Comment (ligfx) [inline CMake/FlagsOverride.cmake]: Why?

File: CMakeLists.txt [hunk 3]
```diff
@@ -9,6 +9,8 @@ set(CMAKE_OSX_ARCHITECTURES "x86_64")
 # features can be used, not the minimum required version to run.
 set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE string "")
 
+set(CMAKE_USER_MAKE_RULES_OVERRIDE "CMake/FlagsOverride.cmake")
```
Comment (ligfx) [inline CMakeLists.txt]: Why do these need to be overridden before the compiler checks?

File: CMake/CCache.cmake (hunk 1 above)
Comment (Orphis) [inline CMake/CCache.cmake]: Because (s)ccache doesn't cache link command and will never do that.

File: CMake/FlagsOverride.cmake [hunk 4]
```diff
@@ -0,0 +1,7 @@
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```
Comment (ligfx) [inline CMake/FlagsOverride.cmake]:
```cmake
if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
  return()
endif()
```
?

File: CMakeLists.txt (hunk 3 above)
Comment (Orphis) [inline CMakeLists.txt]: It's a variable pointing to a script that is run after the compiler checks, it's not run those, but it needs to be set before.

File: CMake/FlagsOverride.cmake (hunk 4 above)
Comment (Orphis) [inline CMake/FlagsOverride.cmake]: We could really use some more overrides on other flags / platform, so this should be fine.

Comment (ligfx) [inline CMake/FlagsOverride.cmake]: 👍

File: CMake/FlagsOverride.cmake (hunk 2 above)
Comment (Orphis) [inline CMake/FlagsOverride.cmake]: No caching tool can save the debug symbols associated with a compilation unit if they are all put in a single PDB file (/Zi). Using /Z7, they are stored in each obj file instead (and then later merged properly by the linker as usual). They will refuse to cache those compilations, so we override the default to use /Z7 instead. It has the potential to reduce contention on that one single file too.

**Comment by ligfx** [comment]
LGTM. I'd love it if a comment with https://stackoverflow.com/a/32956785 could be included somewhere, for explanation, but it works fine as stands.

**Comment by fantesykikachu** [comment]
With cmake going to be used on windows soon what is the status of this PR.

**Pull request merged**